File: cmd/grpc-app/domain/message.go
package domain

import "time"

type Message struct {
	ID        string    `json:"id"`
	Content   string    `json:"content"`
	UserId    string    `json:"userId"`
	CreatedAt time.Time `json:"createdAt"`
	UpdatedAt time.Time `json:"updatedAt"`
	User      User `json:"user"`
}

File: native/lib/domains/message.dart
import 'package:flutter/foundation.dart';
import 'package:freezed_annotation/freezed_annotation.dart';
import 'package:myapp/domains/user.dart';

part 'message.freezed.dart';
part 'message.g.dart';

@freezed
abstract class Message with _$Message {
  const factory Message({
    String id,
    String content,
    String userId,
    DateTime createdAt,
    DateTime updatedAt,
    @JsonKey(name: 'user') User user,
  }) = _Message;

  factory Message.fromJson(Map<String, dynamic> json) => _$MessageFromJson(json);
}
